crates/codegen/src/get_node_properties.rs

```diff
@@ -692,6 +692,74 @@ fn custom_handlers(node: &Node) -> TokenStream {
                 tokens.push(TokenProperty::from(Token::With));
             }
         },
+        "CreatePublicationStmt" => quote! {
+            tokens.push(TokenProperty::from(Token::Create));
+            tokens.push(TokenProperty::from(Token::Publication));
+            if n.for_all_tables {
+                tokens.push(TokenProperty::from(Token::For));
+                tokens.push(TokenProperty::from(Token::All));
+                tokens.push(TokenProperty::from(Token::Tables));
+            }
+            if let Some(n) = n.options.first() {
+                tokens.push(TokenProperty::from(Token::With));
+            }
+            if let Some(n) = n.pubobjects.first() {
+                tokens.push(TokenProperty::from(Token::For));
+                if let Some(NodeEnum::PublicationObjSpec(n)) = &n.node {
+                    match n.pubobjtype() {
+                        protobuf::PublicationObjSpecType::PublicationobjTable => {
+                            tokens.push(TokenProperty::from(Token::Table));
+                        },
+                        protobuf::PublicationObjSpecType::PublicationobjTablesInSchema => {
+                            tokens.push(TokenProperty::from(Token::Tables));
+                            tokens.push(TokenProperty::from(Token::InP));
+                            tokens.push(TokenProperty::from(Token::Schema));
+                        },
+                        _ => panic!("Unknown CreatePublicationStmt {:#?}", n.pubobjtype())
+                    }
+                }
+            }
+            if let Some(n) = n.pubobjects.last() {
+                if let Some(NodeEnum::PublicationObjSpec(n)) = &n.node {
+                    match n.pubobjtype() {
+                        protobuf::PublicationObjSpecType::PublicationobjTablesInSchema => {
+                            tokens.push(TokenProperty::from(Token::Tables));
+                            tokens.push(TokenProperty::from(Token::InP));
+                            tokens.push(TokenProperty::from(Token::Schema));
+                        },
+                        _ => {}
+                    }
+                }
```

Comment:
I think this can be implemented safer. we should do .find() for all types and add the respective tokens if at least one can be found. feel free to fix this @cvng

Reply:
sure, let's have this in a next PR with a `test_create_publication` test